### ISSUE TITLE
Numeric evaluation of Fresnel-type integrals

### ISSUE THREAD

**Issue by normalhuman** [sympy/sympy#15538]
Numeric evaluation of Fresnel-type integrals (increasingly rapid oscillation) is problematic: 
```
>>> import sympy as sym
>>> x = symbols('x')
>>> sym.Integral(sym.sin(x**2), (x, 0, oo)).evalf()
-0.e+104
```
Of course, this is the result of `mpmath.quad` not being able to handle the integral. But if given a suitable hint (zeros of the oscillatory integrand) `quadosc` can do better. 
```
>>> import mpmath as mp
>>> mp.quadosc(lambda x: mp.sin(x**2), [0, mp.inf], zeros=lambda n: mp.sqrt(mp.pi*n))
mpf('0.62665706865775017')
```

This is a long shot, but perhaps SymPy could try to detect oscillation and use `solveset` to produce a `zeros` hint for `quadosc`...  
```
>>> sym.solveset(sym.sin(x**2), x, Interval(0, oo))
Union(ImageSet(Lambda(_n, sqrt(_n)*sqrt(pi)), Integers), ImageSet(Lambda(_n, -sqrt(_n)*sqrt(pi)), Integers))
```
(This part would need to be more accurate.)

**Comment by kesarianubhav** [comment]
@normalhuman  , There is also function  called fresnels (x) , which calculates the fresnel integral of x ,with a normalization factor of  {pie}/2  . !!

**Comment by normalhuman** [comment]
Yes, those are just examples of highly oscillatory Fresnel-type integrals. It could also be an integral of `sin(x**2)*cos(x)` or `cos(x**3)`, etc.

**Comment by asmeurer** [comment]
There is also the `periodicity` function. Though I'm unclear how much symbolic work should be done before applying a numeric function. Could the zeros be found numerically? Does mpmath handle the zeros automatically if none are given?
